compiler/src/iree/compiler/GlobalOptimization/SetEncoding.cpp

```diff
@@ -121,7 +122,7 @@ static MatmulNarrowSizes getMatmulNarrowSizes(ShapedType outType,
   // opportunities to select optimized narrow tiles for narrow matmuls.
   // If it is larger, everything will work fine, but the IR will be a bit more
   // verbose as more narrow_matmul_{M,N} optional parameters will be specified.
-  const int64_t kNarrowThreshold = 16;
+  const int64_t kNarrowThreshold = padFactor ? padFactor : 16;
```

Comment:
Why would padFactor ever be zero ? Trying to drop all magic numbers here. Can't we just use `padFactor` here?

Reply:
I thought it could be zero (a default value?): https://github.com/iree-org/iree/blob/5451e76ca04e700b2386bc01bb5f9c0ca5f4428c/compiler/src/iree/compiler/GlobalOptimization/SetEncoding.cpp#L331

Reply:
and here: https://github.com/iree-org/iree/blob/aef06ed322c2d7789fe1863ff4c884dce6cf03b2/compiler/src/iree/compiler/GlobalOptimization/Passes.h#L119

Reply:
Yes but that doesn't in itself tell whether it's a fundamental possibility or just a local gimmick.

Nevermind, found it:
https://github.com/iree-org/iree/blob/93fc09a2bf14a279f78ce8491958e175bcf8ac6e/compiler/src/iree/compiler/GlobalOptimization/Passes.cpp#L167

So `padFactor = 0` happens when early materialization is enabled (which is default at the moment).

Sorry.. that means that my whole theory of why this `kNarrowThreshold` should use `padFactor`, was wrong. Revert this line to `kNarrowThreshold = 16`, and we'll revisit later how to avoid that hardcoded magic value.

Reply:
yeah.. I'm sorry for introducing the magics. It's easier for testing and connecting e2e functionality; perhaps I made a mistake about the context of padding and tiling. tile_size=0 means nop in tiling. padFactor=0 is UB to me, so I just used the same magic.

I think a better solution could be using `std::optional<int64_t> padFactor`. It's more aligned with how we define the `padFactor` in `EncodingBased.td`. We can also just leave it as what it is, since it is an experimental path for now. (so no action required from my side)